### DISCUSSION TITLE
[MacOS] (Gothic 1) Mouse bug and resolution change

### DISCUSSION THREAD

**Discussion by ghost** [Try/OpenGothic#501] [Q&A] [answered]
Hello,
I tried to run Gothic 1 through OpenGothic and it runs, but there are two things that really interfere with the game.
1. Mouse. When I control the camera through the keyboard, everything is ok, but when I turn on the mouse in the settings, it rises either only up or only down (depending on whether inversion is enabled or not). Because of this, it is impossible to play with the mouse.
2. Change of permission. Neither in the game, nor in the game files, you can change the resolution, which is a serious problem, because. the interface is very small, I literally can't see the text in the dialogs.
Hope you can help me
(Used Google Translate)

**Comment by thokkat** [comment]
> 1.

There was a similar  [issue](https://github.com/Try/OpenGothic/issues/417.)  that has been fixed. Can you compile a build and test it? Or you can try this [build](https://ci.appveyor.com/project/Try/opengothic/builds/47692678/job/yee85j37ikjgtb6q/artifacts) from CI.

> 2.

Still to be implemented. Someone  [changed code](https://github.com/Try/OpenGothic/issues/301#issuecomment-1508944713) for himself to get a proper scaling.

**Comment by Try** [comment]
>  it is impossible to play with the mouse.

Was it already fixed a while ago: https://github.com/Try/OpenGothic/issues/417 ? How old is your build?

> Change of resolution. 

Manipulation system-resolution (specially for no good reason) is not something that we should be doing.
for HiDPI scaling there is a ticket already: https://github.com/Try/OpenGothic/issues/301
for 3D rendering resolution - it's in plans to enable resolution switch here + add a simple upscaler

**Reply by ghost** [reply]
I used this build https://macsourceports.com/game/gothic2

(and thanks for answer about resolution)

**Reply by Try** [reply]
> macsourceports

Yeah, they use very old build - I've poke them a while ago, apparently they haven't made any updates

**Reply by ghost** [reply]
Sorry, but could you give me a little guide on how to run it with a new build? Im bad in this and I don't know how to run new builds

**Reply by Try** [reply, marked as answer]
In principle just build as usual with cmake + XCode - yet, this advice probably not the best for non-developer.
For user end - my suggestion is to reach MacSourcePorts guys and ask them to update the build on their end.

Also as @thokkat mention before, download nightly build is also good option: https://ci.appveyor.com/project/Try/opengothic/builds/47692678/job/yee85j37ikjgtb6q/artifacts

**Reply by ghost** [reply]
Thanks a lot!
